### PR TITLE
ICU-22781 Uncomment and enable constant denominator tests

### PR DESCRIPTION
**Pull request by younies**
# Description: 

Re-enable previously commented out tests for constant denominators in units tests for both ICU4C and ICU4J. This includes tests for:
- Large power of 10 denominators
- Scientific notation denominators
- Specific unit denominators

Added notes about potential CLDR-related edge cases for certain unit representations.

#### Checklist
- [x] Required: Issue filed: ICU-22781
- [x] Required: The PR title must be prefixed with a JIRA Issue number. Example: "ICU-1234 Fix xyz"
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. Example: "ICU-1234 Fix xyz"
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/test/intltest/units_test.cpp [hunk 1]
```diff
@@ -1238,23 +1240,35 @@ void UnitsTest::testUnitsConstantsDenomenator() {
...
         {"meter", 0},
         {"meter-per-second", 0},
         {"meter-per-square-second", 0},
-        // NOTE: The following constant denominator should be 0. However, since
+
+        // NOTE: The following constant denominator should be 1e9. However, since
+        // `portion-per-1e9` is treated as a unit in CLDR,
+        // the unit does not have a constant denominator.
+        // This issue should be addressed in CLDR & ICU.
+        {"portion-per-1000000000", 0},
```
Comment (richgillam) [inline icu4c/source/test/intltest/units_test.cpp:1262]: Is there an outstanding ticket for this?  If so, you should probably include the ticket number (or a link) here.

**Review by richgillam — COMMENTED** [review]
Where's the part where you actually test that formatting does he right thing with this unit?

File: icu4c/source/test/intltest/units_test.cpp (hunk 1 above)
Comment (macchiati) [reply icu4c/source/test/intltest/units_test.cpp:1262]: `portion-per-1e9` is treated as a _precomputed form_ for translation, not a single unit.

[Some units already have 'precomputed' forms, such as **kilometer-per-hour**; where such units exist, they should be used in preference.](https://unicode.org/reports/tr35/tr35-general.html#Unit_Identifier_Uniqueness:~:text=Some%20units%20already%20have%20%27precomputed%27%20forms%2C%20such%20as%20kilometer%2Dper%2Dhour%3B%20where%20such%20units%20exist%2C%20they%20should%20be%20used%20in%20preference.)

We shouldn't test for the wrong behavior. 
That being said, given the schedule, and the fact that we are going fix the problem fully in 48, it's ok for `{"portion-per-1000000000", 1000000000},` be present but commented out.

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java [hunk 2]
```diff
@@ -1344,38 +1345,59 @@ class ConstantDenominatorTestCase {
...
                 new ConstantDenominatorTestCase("meter-per-square-second", 0),
-                // NOTE: The following constant denominator should be 0. However, since
+
+                // NOTE: The following constant denominator should be 1e9. However, since
+                // `portion-per-1e9` is treated as a unit in CLDR,
+                // the unit does not have a constant denominator.
+                // This issue should be addressed in CLDR & ICU.
+                new ConstantDenominatorTestCase("portion-per-1000000000", 0),
+                new ConstantDenominatorTestCase("portion-per-1e9", 0),
+                new ConstantDenominatorTestCase("portion-per-1E9", 0),
```
Comment (macchiati) [inline icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java:1393]: Same here.

**Review by macchiati — CHANGES_REQUESTED** [review]

File: icu4c/source/test/intltest/units_test.cpp [hunk 3]
```diff
@@ -1205,6 +1205,7 @@ void UnitsTest::testUnitsConstantsDenomenator() {
         {"portion-per-7", 7},
         {"portion-per-8", 8},
         {"portion-per-9", 9},
```
Comment (mihnita) [inline icu4c/source/test/intltest/units_test.cpp:1207]: These cases don't exist in the Java tests.
Is there a good reason to have them?
Are they testing something interesting, that is not well covered by other cases?
They they should also be in Java.

If they don't actually exercise anything interesting, we should not have them in C/C++ either.

In general the tests in C++ and Java should be the same, unless there are very good reasons (for example Java testing for all kind of types that don't exist in C++)

File: icu4c/source/test/intltest/units_test.cpp (hunk 1 above)
Comment (younies) [reply icu4c/source/test/intltest/units_test.cpp]: Sure, I have added it back with `logKnownIssue` so it can be addressed in the next release.

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java (hunk 2 above)
Comment (younies) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java]: done, I added also a `logKnownIssue`

File: icu4c/source/test/intltest/units_test.cpp (hunk 3 above)
Comment (younies) [reply icu4c/source/test/intltest/units_test.cpp:1208]: now , c++ and java are identical, sorry for this small confusion.

**Comment by younies** [comment]
> Where's the part where you actually test that formatting does he right thing with this unit?

@richgillam : for testing the formatting:
https://github.com/unicode-org/icu/pull/3389
https://github.com/unicode-org/icu/pull/3390

File: icu4c/source/test/intltest/units_test.cpp [hunk 4]
```diff
@@ -1214,16 +1216,18 @@ void UnitsTest::testUnitsConstantsDenomenator() {
...
+        {"portion-per-1000000000", 1000000000},
+        {"portion-per-10000000000", 10000000000},
+        {"portion-per-100000000000", 100000000000},
+        {"portion-per-1000000000000", 1000000000000},
+        {"portion-per-10000000000000", 10000000000000},
+        {"portion-per-100000000000000", 100000000000000},
+        {"portion-per-1000000000000000", 1000000000000000},
+        {"portion-per-10000000000000000", 10000000000000000},
+        {"portion-per-100000000000000000", 100000000000000000},
+        {"portion-per-1000000000000000000", 1000000000000000000},
```
Comment (richgillam) [inline icu4c/source/test/intltest/units_test.cpp:1228]: Same issue as in the other PRs: If these are all expected to fail and are being covered by `logKnownIssue()` in the code, put some sort of comment next to the actual test case to indicate that (the ticket number you had in the earlier version of these changes works, for example).  Same thing further down in here.

An alternative approach would be to add a `knownIssueTicketNumber` property to your `TestCase` object-- the test code would look for that and, if it's populated, skip the test and pass it to `logKnownIssue`.

**Review by richgillam — COMMENTED** [review]
Looks okay to me, but I think you should get approval from somebody who raised objections the last time around.

File: icu4c/source/test/intltest/units_test.cpp (hunk 4 above)
Comment (richgillam) [reply icu4c/source/test/intltest/units_test.cpp:1228]: Is line 1219 the only one that's failing?  If so, we're good here.

**Comment by younies** [comment]
@macchiati , can you take a look ?

**Review by macchiati — APPROVED** [review]
